pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupRateLimiterTest.java

```diff
@@ -0,0 +1,149 @@
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ResourceGroupRateLimiterTest extends BrokerTestBase {
+
+    final String rgName = "testRG";
+    org.apache.pulsar.common.policies.data.ResourceGroup testAddRg =
+    new org.apache.pulsar.common.policies.data.ResourceGroup();
+    final String namespaceName = "prop/ns-abc";
+    final String topicString = "persistent://prop/ns-abc/test-topic";
+
+    @BeforeClass
+    @Override
+    protected void setup() throws Exception {
+        conf.setMaxPendingPublishRequestsPerConnection(0);
+        super.baseSetup();
+        prepareData();
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    @Override
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    public void createResourceGroup(String rgName, org.apache.pulsar.common.policies.data.ResourceGroup rg) throws PulsarAdminException {
+        admin.resourcegroups().createResourceGroup(rgName, rg);
+
+        Awaitility.await().untilAsserted(() -> {
+            final org.apache.pulsar.broker.resourcegroup.ResourceGroup resourceGroup = pulsar
+                .getResourceGroupServiceManager().resourceGroupGet(rgName);
+            assertNotNull(resourceGroup);
+            assertEquals(rgName, resourceGroup.resourceGroupName);
+        });
+
+    }
+
+    public void deleteResourceGroup(String rgName) throws PulsarAdminException {
+        admin.resourcegroups().deleteResourceGroup(rgName);
+        Awaitility.await().atMost(1, TimeUnit.SECONDS)
+            .untilAsserted(() -> assertNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName)));
+    }
+
+    @Test
+    public void testResourceGroupPublishRateLimit() throws Exception {
+
+        createResourceGroup(rgName, testAddRg);
+        admin.namespaces().setNamespaceResourceGroup(namespaceName, rgName);
+
+        Awaitility.await().untilAsserted(() ->
+            assertNotNull(pulsar.getResourceGroupServiceManager()
+                .getNamespaceResourceGroup(namespaceName)));
+
+        Awaitility.await().untilAsserted(() ->
+            assertNotNull(pulsar.getResourceGroupServiceManager()
+            .resourceGroupGet(rgName).getResourceGroupPublishLimiter()));
+
+        Producer<byte[]> producer = null;
+        try {
+            producer = pulsarClient.newProducer()
+                .topic(topicString)
+                .create();
+        } catch (PulsarClientException p) {
+            final String errMesg = String.format("Got exception while building producer: ex=%s", p.getMessage());
+            Assert.fail(errMesg);
+        }
+
+        MessageId messageId = null;
+        try {
+            // first will be success
+            messageId = producer.sendAsync(new byte[10]).get(100, TimeUnit.MILLISECONDS);
```

Comment:
Nit: have a variable/constant for the '10' (and use the same in the setPublishRateInBytes()?)

Reply:
Done.